Sanitizer error in ydb/tests/fq/streaming_optimize/test_sql_negative.py.test[watermarks-bad_column-default.txt]

ydb/tests/fq/streaming_optimize/test_sql_negative.py.test[watermarks-bad_column-default.txt]

**Build Information:**
- Version: 77b4db1124b682b707bbe3c9cdeecaa7fa6a1b05
- Build Preset: release-asan
- Branch: stable-25-4

**Run Command:**
```
./ya make -ttt --build "release" --sanitize="address" -DDEBUGINFO_LINES_ONLY -F 'test_sql_negative.py::test[watermarks-bad_column-default::txt]' ydb/tests/fq/streaming_optimize
```

**Owner:** [TEAM:@ydb-platform/fq](https://github.com/orgs/ydb-platform/teams/fq)

**Test run history:** [link](https://datalens.yandex/34xnbsom67hcq?branch=stable-25-4&full_name=ydb/tests/fq/streaming_optimize/test_sql_negative.py.test[watermarks-bad_column-default.txt]&build_type=release-asan)

More info in [dashboard](https://datalens.yandex/4un3zdm0zcnyr)

**Logs:**
- [logsdir](https://storage.yandexcloud.net/ydb-gh-logs/ydb-platform/ydb/Regression-run_Small_and_Medium/20903777505/ya-stable-25-4-x86-64-asan/try_1/test_artifacts/ydb-tests-fq-streaming_optimize.zip)

**Error Summary:**
```
ydb/tests/fq/streaming_optimize/test_sql_negative.py:48: in test
    result = fq_run.yql_exec(check_error=False, action="explain")
ydb/tests/fq/tools/fqrun.py:80: in yql_exec
    proc_result = yatest.common.process.execute(cmd.strip().split(), check_exit_code=False, cwd=self.res_dir)
library/python/testing/yatest_common/yatest/common/process.py:656: in execute
    res.wait(check_exit_code, timeout, on_timeout)
library/python/testing/yatest_common/yatest/common/process.py:411: in wait
    self._finalise(check_exit_code)
library/python/testing/yatest_common/yatest/common/process.py:425: in _finalise
    self.verify_sanitize_errors()
library/python/testing/yatest_common/yatest/common/process.py:454: in verify_sanitize_errors
    raise ExecutionError(self)
E   yatest.common.process.ExecutionError: Command '/home/runner/actions_runner/_work/ydb/ydb/tmp/out/ydb/tests/tools/fqrun/fqrun --exclude-linked-udfs --action=explain --cfg=/home/runner/actions_runner/_work/ydb/ydb/tmp/out/ydb/tests/fq/streaming_optimize/test-results/py3test/testing_out_stuff/yql/test_fqrun_et23okke/fq_config.conf --result-file=/home/runner/actions_runner/_work/ydb/ydb/tmp/out/ydb/tests/fq/streaming_optimize/test-results/py3test/testing_out_stuff/yql/test_fqrun_et23okke/results.txt --ast-file=/home/runner/actions_runner/_work/ydb/ydb/tmp/out/ydb/tests/fq/streaming_optimize/test-results/py3test/testing_out_stuff/yql/test_fqrun_et23okke/ast.txt --plan-file=/home/runner/actions_runner/_work/ydb/ydb/tmp/out/ydb/tests/fq/streaming_optimize/test-results/py3test/testing_out_stuff/yql/test_fqrun_et23okke/plan.json --log-file=/home/runner/actions_runner/_work/ydb/ydb/tmp/out/ydb/tests/fq/streaming_optimize/test-results/py3test/testing_out_stuff/yql/test_fqrun_et23okke/log.txt --udfs-dir=/home/runner/actions_runner/_work/ydb/ydb/tmp/out/ydb/tests/fq/streaming_optimize/test-results/py3test/testing_out_stuff/yql_udfs --result-format=full-proto --canonical-output --query=/home/runner/actions_runner/_work/ydb/ydb/tmp/out/ydb/tests/fq/streaming_optimize/test-results/py3test/testing_out_stuff/yql/test_fqrun_et23okke/query_0.sql --emulate-pq=test_topic_input@/home/runner/actions_runner/_work/ydb/ydb/tmp/out/ydb/tests/fq/streaming_optimize/test-results/py3test/testing_out_stuff/yql/test_fqrun_et23okke/topic_0.txt --emulate-pq=test_topic_input2@/home/runner/actions_runner/_work/ydb/ydb/tmp/out/ydb/tests/fq/streaming_optimize/test-results/py3test/testing_out_stuff/yql/test_fqrun_et23okke/topic_1.txt --emulate-pq=test_topic_output@/home/runner/actions_runner/_work/ydb/ydb/tmp/out/ydb/tests/fq/streaming_optimize/test-results/py3test/testing_out_stuff/yql/test_fqrun_et23okke/topic_2.txt --emulate-pq=test_topic_output2@/home/runner/actions_runner/_work/ydb/ydb/tmp/out/ydb/tests/fq/streaming_optimize/test-results/py3test/testing_out_stuff/yql/test_fqrun_et23okke/topic_3.txt' has failed with code 100.
E   Errors:
E   ...mpl /-S/contrib/tools/python3/Python/bltinmodule.c:1093:17
E       #30 0x7b4d3b1cb450 in builtin_exec /-S/contrib/tools/python3/Python/clinic/bltinmodule.c.h:543:20
E       #31 0x7b4d3b14e625 in cfunction_vectorcall_FASTCALL_KEYWORDS /-S/contrib/tools/python3/Objects/methodobject.c:438:24
E       #32 0x7b4d3b1faefd in _PyEval_EvalFrameDefault /tmp/Python/bytecodes.c:3263:26
E       #33 0x7b4d3b0f398a in _PyObject_VectorcallTstate /-S/contrib/tools/python3/Include/internal/pycore_call.h:92:11
E       #34 0x7b4d3b0f398a in object_vacall /-S/contrib/tools/python3/Objects/call.c:850:14
E       #35 0x7b4d3b0f37f8 in PyObject_CallMethodObjArgs /-S/contrib/tools/python3/Objects/call.c:911:24
E       #36 0x7b4d3b24a7dc in import_find_and_load /-S/contrib/tools/python3/Python/import.c:2802:11
E       #37 0x7b4d3b24a7dc in PyImport_ImportModuleLevelObject /-S/contrib/tools/python3/Python/import.c:2885:15
E       #38 0x7b4d3b1ca467 in builtin___import___impl /-S/contrib/tools/python3/Python/bltinmodule.c:276:12
E       #39 0x7b4d3b1ca467 in builtin___import__ /-S/contrib/tools/python3/Python/clinic/bltinmodule.c.h:107:20
E       #40 0x7b4d3b14e625 in cfunction_vectorcall_FASTCALL_KEYWORDS /-S/contrib/tools/python3/Objects/methodobject.c:438:24
E       #41 0x7b4d3b0f2e5c in _PyObject_VectorcallTstate /-S/contrib/tools/python3/Include/internal/pycore_call.h:92:11
E       #42 0x7b4d3b0f2e5c in _PyObject_CallFunctionVa /-S/contrib/tools/python3/Objects/call.c:562:18
E   
E   SUMMARY: AddressSanitizer: 453792 byte(s) leaked in 8617 allocation(s).

logsdir:
/home/runner/actions_runner/_work/ydb/ydb/tmp/out/ydb/tests/fq/streaming_optimize/test-results/py3test/testing_out_stuff
log:
/home/runner/actions_runner/_work/ydb/ydb/tmp/out/ydb/tests/fq/streaming_optimize/test-results/py3test/testing_out_stuff/test_sql_negative.py.test.watermarks-bad_column-default.txt.log
```

--- a/ydb/tests/fq/streaming_optimize/test_sql_negative.py
+++ b/ydb/tests/fq/streaming_optimize/test_sql_negative.py
@@ -31,7 +31,14 @@ def sanitize_issues(s):
     # (yexception) ... ->
     s = re.sub(r"\(yexception\).*", "", s)
     # library/cpp/json/json_reader.cpp:420 -> library/cpp/json/json_reader.cpp:xxx
-    return re.sub(r"cpp:\d+", "cpp:xxx", s)
+    s = re.sub(r"cpp:\d+", "cpp:xxx", s)
+    # Remove DWARF warnings from sanitizer symbolization
+    s = re.sub(r"warning: address range table[^\n]*\n?", "", s)
+    # Remove LSan suppression summary
+    s = re.sub(r"-+\nSuppressions used:\n.*?-+\n?", "", s, flags=re.DOTALL)
+    # Collapse multiple consecutive blank lines into one
+    s = re.sub(r"\n{3,}", "\n\n", s)
+    return s
 
 
 def pytest_generate_tests(metafunc):

--- a/ydb/tests/tools/kqprun/runlib/lsan.supp
+++ b/ydb/tests/tools/kqprun/runlib/lsan.supp
@@ -1,0 +1,6 @@
+leak:Py_InitializeEx
+leak:RegisterYqlPythonUdf
+leak:PyBytes_FromStringAndSize
+leak:PyUnicode_New
+leak:_PyObject_MallocWithType
+leak:init_interp_main

--- a/ydb/tests/tools/kqprun/runlib/ya.make
+++ b/ydb/tests/tools/kqprun/runlib/ya.make
@@ -36,4 +36,8 @@ PEERDIR(
 
 YQL_LAST_ABI_VERSION()
 
+SUPPRESSIONS(
+    lsan.supp
+)
+
 END()

